### DISCUSSION TITLE
Bazel Module: Only first custom registry URL is checked for updates

### DISCUSSION THREAD

**Discussion by PieterBunnenberg** [renovatebot/renovate#25515] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.33.3

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I want to enable renovate to update my Bazel modules. Renovate is able to find the ".bazelrc" and the "MODULE.bazel" file and it extracts the multiple registries. But I got from the logs the following warning:
```
2023-10-26 13:45:47.022941 | ubuntu-2204-2 |  WARN: Excess registryUrls found for datasource lookup - **using first configured only** (repository=org/owner)
2023-10-26 13:45:47.028667 | ubuntu-2204-2 |        "datasource": "bazel",
2023-10-26 13:45:47.028702 | ubuntu-2204-2 |        "packageName": "host-x86_64-linux-llvm15-pkg",
2023-10-26 13:45:47.028723 | ubuntu-2204-2 |        "registryUrls": [
2023-10-26 13:45:47.028729 | ubuntu-2204-2 |          "<registry url 1>",
2023-10-26 13:45:47.028736 | ubuntu-2204-2 |          "<registry url 2>",
2023-10-26 13:45:47.028752 | ubuntu-2204-2 |          "<registry url 3>"
2023-10-26 13:45:47.028758 | ubuntu-2204-2 |        ]
```

I can find the error message for a module that is not stored in the first registry in the log:
```
2023-10-26 13:45:47.299015 | ubuntu-2204-2 | DEBUG: GET <registry url 1>/modules/host-x86_64-linux-llvm15-pkg/metadata.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=273) (repository=org/owner)
```

I cannot find other GET requests to the other registry urls in the log. Hence, I assume that the other registries are not checked. Despite the [doc ](https://docs.renovatebot.com/bazel/#bazel-module-modulebazel-support), all registries should be checked in order of occurrence. This contradicts the warning I get from the log. Modules that are stored in the first registry URL are found and updated. 

Do I miss a configuration entry to enable all registries?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
@cgrindel should Renovate "hunt" these registryUrls instead of us only the first?

It looks like the default value of `first` comes from here: https://github.com/renovatebot/renovate/blob/e231ab88047d52fc3c2870c608656d9c77d3df56/lib/modules/datasource/datasource.ts#L26

And this line is perhaps ignored completely: https://github.com/renovatebot/renovate/blob/e231ab88047d52fc3c2870c608656d9c77d3df56/lib/modules/datasource/index.ts#L285

**Comment by cgrindel** [comment]
This looks like a bug in the `BazelDataSource`. I will put up a PR to ensure that the `registryStrategy` is set in the data source.

**Comment by cgrindel** [comment]
@PieterBunnenberg Can you see if [the changes in this PR](https://github.com/renovatebot/renovate/pull/25520) fix your problem?

**Comment by PieterBunnenberg** [comment, marked as answer]
@cgrindel Thank you very much for your fast response and fix. I can confirm that the warning is gone and alle registries are checked for updates. 🎉 I tested with version: 37.42.0
